forward/forwarder.go

```diff
@@ -50,6 +50,7 @@ type Options struct {
 	RerouteRetries bool
 	RetrySchedule  []time.Duration
 	Timeout        time.Duration
+	Headers        []byte
```

Comment:
Headers in tchannel are represented as a map of strings:
in `tchannel-go/context_header.go`:
```
30     Headers() map[string]string
```

Maybe we should expose a data structure similar to tchannel's?